### PR TITLE
feat: Update for new `run_multiple` API and internal Julia changes

### PR DESCRIPTION
**Pull request by kshyatt-aws**
*Issue #, if available:* N/A

*Description of changes:* set up pass-through for `run_multiple` to conform to new method and use `_map_contiguous_qubits` appropriately

*Testing done:* `tox` passed locally

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-simulator-v2-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-simulator-v2-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/amazon-braket/amazon-braket-simulator-v2-python/blob/main/README.md) and [API docs](https://github.com/amazon-braket/amazon-braket-simulator-v2-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/braket/simulator_v2/base_simulator_v2.py [hunk 1]
```diff
@@ -1,7 +1,10 @@
-from abc import ABC
+import sys
+import warnings
+from collections.abc import Sequence
+from typing import Any, Optional, Union
 
+import juliacall
```
Comment (speller26) [inline src/braket/simulator_v2/base_simulator_v2.py:6]: Is this import used at all? I only see `JuliaError`

File: src/braket/simulator_v2/base_simulator_v2.py [hunk 2]
```diff
@@ -44,46 +93,48 @@ def run_jaqcd(
...
-                    result_type
-                    for result_type in result_types
-                    if isinstance(result_type, TargetedResultType)
-                ],
-                qubit_count,
-            )
-        r = jl.simulate(self._device, [circuit_ir], qubit_count, shots)
+        translated_ir, qubit_count = self._jaqcd_to_jl(circuit_ir, shots)
+        try:
+            r = jl.simulate(self._device, translated_ir, qubit_count, shots)
```
Comment (speller26) [inline src/braket/simulator_v2/base_simulator_v2.py:102]: Does the underlying Julia simulator need `qubit_count`? If not, can we remove this parameter (and remove the `qubit_count` return altogether from `_jaqcd_to_jl`.

File: src/braket/simulator_v2/base_simulator_v2.py [hunk 3]
```diff
@@ -105,54 +156,94 @@ def run_openqasm(
...
-        operations = circuit.instructions
-        BaseLocalSimulator._validate_operation_qubits(operations)
+        for r_ix, result in enumerate(results):
+            results[r_ix].additionalMetadata.action = programs[r_ix]
+            # attach the result types
+            if not shots:
+                results[r_ix] = _result_value_to_ndarray(result)
+            else:
+                if isinstance(programs[r_ix], OpenQASMProgram):
+                    results[r_ix].resultTypes = [rt.type for rt in result.resultTypes]
```
Comment (speller26) [inline src/braket/simulator_v2/base_simulator_v2.py:222]: Why is this so different from the others? And can the block be changed to
```
if isinstance(programs[r_ix], OpenQASMProgram) and shots:
    ...
else:
    ...
```

**Review by speller26 — APPROVED** [review]
Optional cosmetic comments

File: src/braket/simulator_v2/base_simulator_v2.py (hunk 2 above)
Comment (kshyatt-aws) [reply src/braket/simulator_v2/base_simulator_v2.py:102]: It doesn't strictly need it but it does help not having to recompute it

**Comment by speller26** [comment]
This feels like a big enough change to warrant `feat: `

**Pull request merged**